lib/db_connection.ex

```diff
@@ -80,15 +80,19 @@ defmodule DBConnection do
 
   defstruct [:pool_mod, :pool_ref, :conn_mod, :conn_ref, :conn_mode]
 
+  defmodule EncodeError do
+    defexception [:message]
```

Comment:
Do we want to support `:value` for the term that was failed to be encoded? For Ecto this is the wrong place to handle this though so don't think we need it.